llvm/lib/Target/DirectX/DXIL.td

```diff
@@ -108,55 +96,55 @@ class DXILOperation<string name, int opCode, DXILOpClass opClass, DXILOpCategory
 class LLVMIntrinsic<Intrinsic llvm_intrinsic_> { Intrinsic llvm_intrinsic = llvm_intrinsic_; }
 
 def Sin : DXILOperation<"Sin", 13, UnaryClass, UnaryFloatCategory, "returns sine(theta) for theta in radians.",
-  [f16Ty,f32Ty], ReadNone,
+  [llvm_half_ty,llvm_float_ty], ReadNone,
   [
-    DXILOpParameter<0, "$o", "", "operation result">,
-    DXILOpParameter<1, "i32", "opcode", "DXIL opcode">,
-    DXILOpParameter<2, "$o", "value", "input value">
+    DXILOpParameter<0, llvm_anyfloat_ty, "", "operation result">,
+    DXILOpParameter<1, llvm_i32_ty, "opcode", "DXIL opcode">,
+    DXILOpParameter<2, llvm_anyfloat_ty, "value", "input value">
```

Comment:
The $o for parameter 2 needs to be the same as parameter 0,
Is it possible to get something like LLVMMatchType?

Reply:
> The $o for parameter 2 needs to be the same as parameter 0, Is it possible to get something like LLVMMatchType?

Good idea. Thanks!

However, I will need to investigate how such a constraint would be enforced in the backend code (??). Additionally, it appears that parameter types are expected to be specified as a list of `LLVMTypes`. 

@python3kgae, would you be OK if the change you suggested is done in a subsequent PR?